transportable-udfs-trino/src/main/java/com/linkedin/transport/trino/StdUdfWrapper.java

```diff
@@ -98,8 +106,34 @@ protected long getRefreshIntervalMillis() {
   }
 
   @Override
-  public ScalarFunctionImplementation specialize(BoundVariables boundVariables, int arity, Metadata metadata) {
-    StdFactory stdFactory = new PrestoFactory(boundVariables, metadata);
+  public FunctionDependencyDeclaration getFunctionDependencies(FunctionBinding functionBinding) {
+    FunctionDependencyDeclaration.FunctionDependencyDeclarationBuilder builder = FunctionDependencyDeclaration.builder();
+
+    List<Type> types = new ArrayList<>();
+    types.add(functionBinding.getBoundSignature().getReturnType());
+    types.addAll(functionBinding.getBoundSignature().getArgumentTypes());
+
+    types.forEach(type -> builder.addType(type.getTypeSignature()));
+    if (containsMapType(types)) {
+      builder.addOperatorSignature(EQUAL, ImmutableList.of(new TypeSignature("K"), new TypeSignature("K")));
```

Comment:
Is the type signature being hardcoded to `K` correct? Shouldn't it apply to the type of the map key? What if there are multiple map types with different map key types?

Reply:
good point! updated it to the map key type

Reply:
I am not sure if this is fully correct. For nested types we may need to recurse through the type signatures to identify map keys within struct types. Is there are a good way we can test this?

Reply:
Can you share an example of this?

Reply:
e.g. a type like with `struct<field:map<K,V>>`

Reply:
registered nested types as well.